### PR TITLE
[UIE-136] Build test utils package

### PR DESCRIPTION
**Pull request by nawatts**
This adds a build step to the test-utils package.

This brings test-utils more in line with core-utils and components, adds TypeScript support, and sets us up to use test-utils for shared helpers used within tests as well as shared Jest configuration. As an example of that, this moves `withFakeTimers` from components' internal test-utils module to the test-utils package.

To allow adding test helpers, I had to change test-utils to export a function to get the Jest configuration instead of the Jest configuration object itself. Otherwise, tests would fail when importing from `@terra-ui-packages/test-utils`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -67,7 +67,7 @@
     "analyze": "yarn build && source-map-explorer 'build/static/js/*.js' --gzip",
     "build": "yarn build-packages && yarn build-app",
     "build-app": "tsc && yarn save-build-info && vite build && rm build/config.json",
-    "build-packages": "yarn workspaces foreach --include '@terra-ui-packages/*' --parallel run build",
+    "build-packages": "yarn workspaces foreach --include '@terra-ui-packages/*' --parallel --topological-dev run build",
```
Comment (nawatts) [inline package.json:70]: Packages must now be built in order based on their interdependencies.
https://yarnpkg.com/cli/workspaces/foreach#details

File: jest.config.mjs [hunk 2]
```diff
@@ -1,16 +1,21 @@
-import { baseConfig } from '@terra-ui-packages/test-utils';
+import { getJestConfig } from '@terra-ui-packages/test-utils';
+
+const baseConfig = getJestConfig();
 
 export default {
   ...baseConfig,
-  'reporters': [
```
Comment (nawatts) [inline jest.config.mjs:5]: Formatted this file while I was here.

**Review by msilva-broad — APPROVED** [review]
looks good to me.

happy to see this being promoted to full build goodness.

File: packages/test-utils/package.json [hunk 3]
```diff
@@ -1,16 +1,28 @@
 {
   "name": "@terra-ui-packages/test-utils",
-  "version": "0.0.2",
+  "version": "0.0.4",
```
Comment (cahrens) [inline packages/test-utils/package.json]: Any reason you are skipping 0.0.3?

**Review by cahrens — APPROVED** [review]

File: packages/test-utils/package.json (hunk 3 above)
Comment (nawatts) [reply packages/test-utils/package.json]: Oops, looks like I bumped the version in two commits within this PR.

Comment (nawatts) [reply packages/test-utils/package.json]: Updated in https://github.com/DataBiosphere/terra-ui/pull/4294/commits/746f35d586efdbd07c4f01f8241f34faf9cf698d.

**Pull request merged**